### DISCUSSION TITLE
Current state of islands and islandsRouter?

### DISCUSSION THREAD

**Discussion by Dindaleon** [solidjs/solid-start#730] [Q&A] [answered]
I am starting to work with solidjs and solid-start and tried to activate islands and islandsRouter but my app broke, not even a basic counter works. So i was wondering if this feature is currently working and if so, is there any demo or tutorial I can follow to see what I am doing wrong?

**Comment by boehs** [comment, marked as answer]
You need to explicitly opt into islands, like `const Counter = unstable_island(() => import('./components/counter'))`

**Reply by Dindaleon** [reply]
ok cool, is it in the docs? I would like to know more about it.

**Reply by boehs** [reply]
> ok cool, is it in the docs? I would like to know more about it.

As implied, islands are unstable

**Reply by mb21** [reply]
As of `solidjs/start 1.0.8` this seems to have changed? [The docs](https://docs.solidjs.com/solid-start/reference/config/define-config) mention `experimental: { islands: true }}`, but when I put a `'use client';` on the first line of `Counter.tsx` in a `basic`-template-initialized project I get this in the browser:

    Uncaught (in promise) Error: Hydration Mismatch. Unable to find DOM nodes for hydration key: 0011-00

Is this currently broken or am I doing something wrong? Would love to experiment around with experimental islands.

Edit: oh, it actually works in the production build, just not in the dev server.
